pkgs/gnolang/precompile.go

```diff
@@ -190,6 +190,10 @@ func PrecompileBuildPackage(fileOrPkg string, goBinary string) error {
 	sort.Strings(files)
 	args := append([]string{"build", "-v", "-tags=gno"}, files...)
 	cmd := exec.Command(goBinary, args...)
+	rootDir, err := guessRootDir(fileOrPkg, goBinary)
+	if err == nil {
+		cmd.Dir = rootDir
+	}
```

Comment:
Can you please explain the logic behind this error handling?

Why not return an error here, but instead modify the `cmd.Dir` value?

Reply:
I wanted to preserve the original flow here. Only change the cmd directory if we were able to guess the root directory otherwise do nothing.